plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/JavaTextDocumentServiceWraper.java

```diff
@@ -36,4 +63,45 @@ public JavaTextDocumentServiceWraper(TextDocumentService wrapped) {
           return commands;
         });
   }
+
+  public CompletableFuture<List<? extends TextEdit>> formatting(DocumentFormattingParams params)
+      throws LanguageServerException {
+
+    String fileUri = params.getTextDocument().getUri();
+    FormattingOptions options = params.getOptions();
+    if (options == null) {
+      options = new FormattingOptions();
+      params.setOptions(options);
+    }
+
+    String projectPath = WsPathUtils.absolutize(extractProjectPath(fileUri));
```

Comment:
It would be nice to have a private method to update formatter options

Reply:
ok